### ISSUE TITLE
Improve error logging for the `infer` warning in schema customization

### ISSUE THREAD

**Issue by jlengstorf** [gatsbyjs/gatsby#19314]
Right now, a warning gets thrown when a type is created when `@dontInfer` and/or `infer: false` is set on a custom type, but custom fields are added anyways.

The error looks like this:

```
warn On types with the `@dontInfer` directive, or with the `infer`
extension set to `false`, automatically adding fields for children
types is deprecated.
```

This is great, but it would be _really_ helpful if this warning mentioned _which type_ is causing the warning. Bonus points if it said which field(s) are being automatically added.

For example:

```
warn On types with the `@dontInfer` directive, or with the `infer`
extension set to `false`, automatically adding fields for children
types is deprecated. The type `YourType` does not explicitly define
the field `someData`.
```

This could potentially be part of #18873, but maybe this could go in earlier on its own to help site authors clean up their schemas before things start breaking?

Thanks team!

**Comment by vladar** [comment]
The interesting thing is that we already have this information in the warning:

https://github.com/gatsbyjs/gatsby/blob/393c1f0236f154bbf2e303789d88f242c29e7327/packages/gatsby/src/schema/schema.js#L943-L950

But I saw several cases when warnings were cut off from below. Maybe it is actually a bug with gatsby `ink` reporter. Could try it with `yurnalist` please:

```
GATSBY_LOGGER=yurnalist gatsby build
```

Curious if it produces the same result for you?

**Comment by jlengstorf** [comment]
@vladar maybe invert the instructions, then?

```
warn The type `YourType` does not explicitly define the field 
`someData`. On types with the `@dontInfer` directive, or with 
the `infer` extension set to `false`, automatically adding 
fields for children types is deprecated.
```

Even if it gets cut off, then, the most important information is still visible.

**Comment by vladar** [comment]
Sounds more clear, thanks! Will prepare a PR for this. Also, I've figured that the root cause of the issue is indeed related to `ink`, created a bug report for them - https://github.com/vadimdemedes/ink/issues/245

**Comment by vladar** [comment]
Published in `gatsby 2.18.2`
